addons/l10n_latam_check/wizards/account_payment_register.py

```diff
@@ -42,3 +50,15 @@ def _create_payment_vals_from_wizard(self, batch_result):
                 'l10n_latam_move_check_ids': [Command.link(x.id) for x in self.l10n_latam_move_check_ids]
             })
         return vals
+
+    def action_create_payments(self):
+        if self._is_latam_check_payment(check_subtype="move_check") and (
+            len(self.l10n_latam_move_check_ids.mapped("currency_id")) > 1
+            or self.l10n_latam_move_check_ids[:1].currency_id != self.currency_id
+        ):
+            raise ValidationError(_(
+                "You can't mix checks of different currencies in one payment, "
+                "and you can't change the payment's currency if checks are already created in that currency.\n"
+                "Please create separate payments for each currency."
+            ))
+        return super().action_create_payments()
```

Comment:
I think now it will raise if `self.l10n_latam_move_check_ids` is empty. Maybe we can do something like this instead?

```suggestion
        if not self._is_latam_check_payment(check_subtype="move_check"):
            return super().action_create_payments()

        latam_check_currencies = self.l10n_latam_move_check_ids.mapped("currency_id")
        if latam_check_currencies and (len(latam_check_currencies) > 1 or latam_check_currencies != self.currency_id):
            raise ValidationError(_(
                "You can't mix checks of different currencies in one payment, "
                "and you can't change the payment's currency if checks are already created in that currency.\n"
                "Please create separate payments for each currency."
            ))
        return super().action_create_payments()
```